### ISSUE TITLE
Tempo TraceQL queries do not interpolate dashboard variables

### ISSUE THREAD

**Issue by dggreenbaum** [grafana/grafana#63745]
**What happened**:

I tried to populate a Table panel with the results of a TraceQL query. I attempted to use the [Variable Syntax](https://grafana.com/docs/grafana/latest/dashboards/variables/variable-syntax/) to feed in the selected value from a dashboard variable. When the query was executed the literal value of my query was executed rather than what I expected the interpolated output to be.

![image](https://user-images.githubusercontent.com/721195/221226124-e54914b9-3e29-4a96-98d4-1518fde2ff5b.png)

![image](https://user-images.githubusercontent.com/721195/221226340-6a7e0602-52c5-4735-943a-0d4e2ca44f68.png)

**What you expected to happen**:

I epected the following query to be executed

```
{resource.service.name = "foo"  && status = error} 
```

Instead this query was executed

```
{resource.service.name = "${service}"  && status = error} 
```

**How to reproduce it (as minimally and precisely as possible)**:

- Create a Grafana dashboard with a query or custom variable defined with 1 or more selectable options
- Create a new Table panel with a Tempo TraceQL query
- Define a TraceQL query where you use the [Variable Syntax](https://grafana.com/docs/grafana/latest/dashboards/variables/variable-syntax/) to read in the value of the above dashboard variable
- Inspect the query via the Query Inspector interface to see that dashboard variable reference is sent literally to the query backend

**Environment**:
- Grafana version: v9.3.6 (978237e7cb) ([DockerHub link](https://hub.docker.com/layers/grafana/grafana/9.3.6/images/sha256-825065ced6a481f088d53b1c42fdfb9beadce6f87a07af614be1b1a78461e2bd?context=explore))
- Data source type & version: Tempo 2.0.0
- OS Grafana is installed on: Ubuntu 20.04 (Docker)
- User OS & Browser: Chrome
- Grafana plugins: None

**Comment by mellieA** [comment]
@dggreenbaum I tried this out and it seems to work as expected on 9.4.1 but I'm just using a custom variable. Can you share more about how your variable is defined? 

![Screenshot 2023-03-01 at 2 34 30 PM](https://user-images.githubusercontent.com/18175287/222270001-8cd5a879-9698-4931-9089-c16c097300e3.png)

**Comment by nijave** [comment]
Hitting this on Grafana Cloud 9.3.6. Sounds like a problem in 9.3 that was fixed in 9.4 https://community.grafana.com/t/using-dashboard-variable-inside-traceql/81395/8

**Comment by nijave** [comment]
https://github.com/grafana/grafana/pull/60414

**Comment by dggreenbaum** [comment]
Confirmed this is fixed in v9.4. Thanks!

**Comment by dggreenbaum** [comment]
@mellieA the variable in my example is a custom variable I defined as a static list. In the production dashboard I noticed this in it's populated by a label_values query from Mirmr. Both behave the same on v9.3 and work as expected in v9.4.
